### PR TITLE
refactor: Refactor analyzer config implementation to match Roslyn

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): #

## PR Type
What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno.SourceGeneration/tree/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] Tests for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno.SourceGeneration/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by Youssef1313** [comment]
@jeromelaban This reverted the way I wrote the code and ported it from Roslyn to match their code instead.

**Comment by jeromelaban** [comment]
Is there are behavior different to test for?

**Comment by Youssef1313** [comment]
@jeromelaban I wasn't able to notice a difference. But since this code is copied from Roslyn, it's better to keep in sync.

File: src/Uno.SourceGeneration.Engine.Shared/Workspace/CommandLineArgumentReader.cs [hunk 1]
```diff
@@ -40,11 +42,11 @@ protected void Add(string name)
             _builder.Add($"/{name}");
         }
 
-        protected void Add(string name, string value)
+        protected void Add(string name, string? value, bool addQuoteIfValueContainsWhitespace = true)
```
Comment (Youssef1313) [inline src/Uno.SourceGeneration.Engine.Shared/Workspace/CommandLineArgumentReader.cs:45]: This was a bug fixed in https://github.com/dotnet/roslyn/pull/57117

File: src/Uno.SourceGeneration.Engine.Shared/Workspace/ProjectFileInfo.cs [hunk 2]
```diff
@@ -44,6 +46,19 @@ internal sealed class ProjectFileInfo
...
+        // /// or null if it is unknown or not applicable. 
+        // /// </summary>
+        // /// <remarks>
+        // /// Right now VB doesn't have the concept of "default namespace". But we conjure one in workspace 
+        // /// by assigning the value of the project's root namespace to it. So various feature can choose to 
+        // /// use it for their own purpose.
+        // /// In the future, we might consider officially exposing "default namespace" for VB project 
+        // /// (e.g. through a "defaultnamespace" msbuild property)
+        // /// </remarks>
+        // public string DefaultNamespace { get; }
```
Comment (Youssef1313) [inline src/Uno.SourceGeneration.Engine.Shared/Workspace/ProjectFileInfo.cs:60]: I synced the code from Roslyn and commented out things we're not using.

Comment (Xiaoy312) [reply src/Uno.SourceGeneration.Engine.Shared/Workspace/ProjectFileInfo.cs:60]: for large chunk, you can wrap them in `#if false`

Comment (Xiaoy312) [reply src/Uno.SourceGeneration.Engine.Shared/Workspace/ProjectFileInfo.cs:60]: for large chunk, you can wrap them in `#if false`

**Comment by jeromelaban** [comment]
The builld is failing on incompatible newer features

**Comment by Youssef1313** [comment]
@jeromelaban This is green now

**Review by jeromelaban — APPROVED** [review]
Thanks!

**Pull request merged**